spec/system/service_configuration_spec.rb

```diff
@@ -278,7 +291,148 @@ def batlight_running_on_instance(ip)
             _=$(killAndAwaitProcess bosh-agent)
             echo "SUCCESS"
           EOF
-          output = bosh_ssh('batlight', 0, cmd, deployment: deployment.name).output
+          output = bosh_ssh(instance_name, instance_id, cmd, deployment: deployment.name).output
+          expect(output).to include("SUCCESS")
+        end
+      end
+    end
+  end
+
+  describe 'systemd' do
+    before(:all) do
+      if srv_cmd == "sv"
+        skip 'Not applicable for agent/monit running on runit'
+      end
+    end
+
+    after(:each) do
+      instance_reboot
+    end
+
+    context 'when initially started after instance boot (before agent has been started)' do
+      # it 'deletes /etc/service/monit' do
+      #   # expect /etc/service/monit to be younger that the system's uptime
+      #   cmd = <<-EOF
+      #     #{bash_functions}
+      #     _=$(waitForSymlink /etc/service/monit)
+      #     now="$(date +"%s")"
+      #     mod_time="$(stat --printf="%Y" /etc/service/monit)"
+      #     up_time="$(cut -f1 -d. /proc/uptime)"
+      #     diff=$((${up_time}-${now}+${mod_time}))
+      #     if [ $diff -ge 0 ]; then
+      #       echo "SUCCESS"
+      #     else
+      #       echo "FAILURE: expected /etc/service/monit to be younger than uptime, got a difference of ${diff} seconds"
+      #       exit 1
+      #     fi
+      #   EOF
+      #   expect(ssh(public_ip, 'vcap', cmd, ssh_options(@spec))).to include("SUCCESS\n")
+      # end
+
+      context 'when monit dies' do
+        it 'restarts it' do
+          # compare monit pids pre- and post kill
+          cmd = <<-EOF
+            #{bash_functions}
+            old_pid="$(waitForProcess monit "")"
+            sudo kill ${old_pid}
+            new_pid="$(waitForProcess monit $old_pid)"
+            if [[ "${new_pid}" = "${old_pid}" || -z "${new_pid}" ]]; then
+              echo "FAILURE"
+              exit 1
+            fi
+            echo "SUCCESS"
+          EOF
+          output = bosh_ssh(instance_name, instance_id, cmd, deployment: deployment.name).output
+          expect(output).to include("SUCCESS")
+        end
+      end
+
+      context 'when the agent dies' do
+        it 'restarts it' do
+          # compare agent pids pre- and post kill
+          cmd = <<-EOF
+            #{bash_functions}
+            old_pid="$(waitForProcess bosh-agent "")"
+            sudo kill ${old_pid}
+            new_pid="$(waitForProcess bosh-agent $old_pid)"
+            if [[ "${new_pid}" = "${old_pid}" || -z "${new_pid}" ]]; then
+              echo "FAILURE"
+              exit 1
+            fi
+            echo "SUCCESS"
+          EOF
+          output = bosh_ssh(instance_name, instance_id, cmd, deployment: deployment.name).output
+          expect(output).to include("SUCCESS")
+        end
+      end
+    end
+
+    context 'when restarted after agent has been started' do
+      # it 'does not delete /etc/service/monit' do
```

Comment:
Can we still do this on sv machines? for jammy stemcells

Reply:
yes i need to put an systemd condition for that as well.